### PR TITLE
[Geometry] Add methods getBarycentricCoordinates and isPointInTriangle in Triangle class

### PR DESCRIPTION
**Pull request by epernod**
Add method:
- pointBaryCoefs to Compute the barycentric coefficients of input point in the current Triangle
- method isPointInTriangle  using the barycentric coefficients
- Add unit tests with positive and negative cases

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: Sofa/framework/Geometry/src/sofa/geometry/Triangle.h [hunk 1]
```diff
@@ -71,6 +71,44 @@ struct Triangle
...
 
+
+    /**
+    * @brief	Compute the barycentric coefficients of input point on Edge (n0, n1)
+    * @tparam   Node iterable container
+    * @tparam   T scalar
+    * @param	point: position of the point to compute the coefficients
+    * @param	n0,n1: nodes of the edge
+    * @return	sofa::type::Vec<2, T> barycentric coefficients
+    */
```
Comment (alxbilger) [inline Sofa/framework/Geometry/src/sofa/geometry/Triangle.h]: The documentation is not appropriate for a triangle. I guess this was a copy/paste from `pointBaryCoefs` in `Edge.h`

**Comment by epernod** [comment]
> Do you know why the function is not removed in the other code? We maybe should create an issue not to forget to clean the GeometryAlgorithms

For the moment, the old method is not removed because it was not well handling all cases. Therefor the new method has a slightly different (better) behavior. Switching to the new one might cause some scene regressions to fail. I prefer to do that in a next PR. 

For the TriangleSetGeometryAlgorithm clean, you can create an issue if you want but it is already a wip on my dev branch: See https://github.com/epernod/sofa/pull/10/files#diff-e1443ad8032ebc3d01202980477d0fd24cf3c4db16a90452dd51f780bffe2e3f
I will push the changes progressively

File: Sofa/framework/Geometry/src/sofa/geometry/Triangle.h [hunk 2]
```diff
@@ -71,6 +71,45 @@ struct Triangle
...
+    * @tparam   T scalar
+    * @param	p0: position of the input point to compute the coefficients
+    * @param	n0, n1, n2: nodes of the triangle
+    * @return	sofa::type::Vec<3, T> barycentric coefficients 
+    */
+    template<typename Node,
+        typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
+        typename = std::enable_if_t<std::is_scalar_v<T>>
+    >
+        static constexpr auto pointBaryCoefs(const Node& p0, const Node& n0, const Node& n1, const Node& n2)
```
Comment (fredroy) [inline Sofa/framework/Geometry/src/sofa/geometry/Triangle.h]: `pointBaryCoefs`, the name looks... not really informative and uses abbreviations.
Maybe something like `getBarycentricCoefficients()` or even `getBarycentricCoordinates()` which seems more in par with the maths

**Review by fredroy — CHANGES_REQUESTED** [review]

File: Sofa/framework/Geometry/src/sofa/geometry/Triangle.h (hunk 2 above)
Comment (epernod) [reply Sofa/framework/Geometry/src/sofa/geometry/Triangle.h]: Done and add more description on the method (based from wikipedia)

**Review by fredroy — APPROVED** [review]
